keyboards/dztech/dz65rgb/keymaps/drootz/keymap.c

```diff
@@ -0,0 +1,774 @@
+static uint8_t  fade_in_step_counter, 
+                fade_out_step_counter, 
+                recording_r, 
+                recording_g, 
+                recording_b;
+
+/**************** HELPER FUNCTIONS *********************/
+
+uint8_t get_blink_rgb(uint16_t h, uint16_t s, uint16_t v, uint8_t output) {
+    HSV hsv = {h, s, v * ((ledDimRatio / 2) + ledDimRatio)};
+    RGB rgb = hsv_to_rgb(hsv);
+    const uint8_t HSV_RGB[3] = {rgb.r, rgb.g, rgb.b};
+    switch (output) {
+        case 0:
+            return HSV_RGB[0];
+            break;
+        case 1:
+            return HSV_RGB[1];
+            break;
+        case 2:
+            return HSV_RGB[2];
+            break;
+        default:
+            return false;
+    }
+}
+
+void set_blink_rgb(uint16_t h, uint16_t s, uint16_t v) {
+    recording_r = get_blink_rgb(h, s, v, 0);
+    recording_g = get_blink_rgb(h, s, v, 1);
+    recording_b = get_blink_rgb(h, s, v, 2);
+}
+
+void reset_blink_cycle(void) {
+    blink_cycle_timer = timer_read();
+    blink_fade_in_timer = timer_read();
+    blink_fade_out_timer = timer_read();
+    blink_hsv_value = 0;
+    fade_in_step_counter = 0;
+    fade_out_step_counter = 0;
+}
+
+void reset_blink_status(void) {
+    isRecording = false;
+    isPlaying = false;
+    isBlinking = false;
+}
+
+void get_this_led_blinking(uint16_t led_index, bool speed, uint16_t hue, uint16_t sat) {
+    const uint16_t static_on_time = speed ? 200 : 500;
+    const uint16_t static_off_time = speed ? 200 : 500;
+    const uint8_t fade_timing = speed ? 100 : 150;
+    const uint8_t fade_step = speed ? 10 : 15;
+    const uint8_t fade_value_decrement_size = rgb_matrix_config.hsv.v / fade_step;
+    const uint8_t fade_cycle_time_elapsed = fade_timing / fade_step;
+    if (timer_elapsed(blink_cycle_timer) < static_on_time) {
+        if (timer_elapsed(blink_fade_in_timer) > fade_cycle_time_elapsed && fade_in_step_counter < fade_step) {
+            blink_hsv_value = blink_hsv_value + fade_value_decrement_size;
+            set_blink_rgb(hue, sat, blink_hsv_value);
+            fade_in_step_counter = fade_in_step_counter + 1;
+            blink_fade_in_timer = timer_read();
+        }
+    } else {
+        if (timer_elapsed(blink_fade_out_timer) > fade_cycle_time_elapsed && fade_out_step_counter < fade_step) {
+            blink_hsv_value = blink_hsv_value - fade_value_decrement_size;
+            set_blink_rgb(hue, sat, blink_hsv_value);
+            fade_out_step_counter = fade_out_step_counter + 1;
+            blink_fade_out_timer = timer_read();
+        }
+    }
+    if (timer_elapsed(blink_cycle_timer) > static_on_time + static_off_time) {
+        reset_blink_cycle();
+    }
+    rgb_matrix_set_color(led_index, recording_r, recording_g, recording_b);
+}
+
+/* Mac require a short delay before Capslock register in macros */
+void registerCapsLock(void) {
+    register_code(KC_CAPSLOCK);
+    wait_ms(80);
+    unregister_code(KC_CAPSLOCK);
+}
```

Comment:
You can just use `tap_code` for capslock. 

Just add `#define TAP_HOLD_CAPS_DELAY 80` to your config.h. 

https://github.com/qmk/qmk_firmware/blob/c3200aa8fcf86122736846017ca7b125ac225842/tmk_core/common/action.c#L874-L882

Reply:
Thanks! Addressed in PR #8199 Feedback Commit #1 - c0be53d